app/controllers/user_controller.py

```diff
@@ -0,0 +1,18 @@
+from typing import Tuple
+from flask import Blueprint
+from app.models import User
+from app.utils.session_decorators import session_required
+
+
+user_routes = Blueprint('user_routes', __name__)
+
+
+@user_routes.route('/<username>')
+@session_required
+def get_user(username) -> Tuple[str, int]:
+    user = User.get_by_username(username)
+    if user:
+        return '', 200
+    else:
+        return '', 204
```

Comment:
For the body something like this. The endpoint isn't a json endpoint it will be a 'get' so you have to return the template (which isnt created yet so call it whatever you want)

They will also need on the profile page if the current user is the logged in user (so they will show the delete post button and will hide the follow/unfollow button) so you can pull in the user_id from the @session_required wrapper and the do a comparison with the user obj you fetched.

Below is a general idea of what it would look like

```suggestion
def get_user(user_id, username) -> Any:
    if request.method == 'GET':
        user = User.get_by_username(username)
        is_logged_in_user = user.user_id = user_id
        if user:
            return render_template('user_profile.html', user=user, is_logged_in_user=is_logged_in_user)
        else:
            return '', 404
```

Reply:
I wanted to let rendering the template for the FE guys. I'll added the code.